server/src/main/java/org/elasticsearch/repositories/IndexMetaDataGenerations.java

```diff
@@ -193,4 +211,30 @@ public static String buildUniqueIdentifier(IndexMetadata indexMetaData) {
             + "-"
             + indexMetaData.getAliasesVersion();
     }
+
+    /**
+     * Given a blobId, returns the index UUID associated with it.
+     * <p>
+     * If the blobId is not found, returns null.
+     * <p>
+     * If the index UUID is _na_ {@code ClusterState.UNKNOWN_UUID}
+     * @param blobId The blob ID
+     */
+    @Nullable
+    public String getIndexUUIDFromBlobId(String blobId) {
+        // Find the unique identifier for this blobId
+        String uniqueIdentifier = blobUuidToIndexMetadataMap.get(blobId);
+        if (uniqueIdentifier == null) {
+            return null;
+        }
+
+        // The uniqueIdentifier is built in {@code buildUniqueIdentifier}, and is prefixed with indexUUID
+        // The indexUUID is either a UUID of length 22, or _na_
+        boolean na = uniqueIdentifier.startsWith(ClusterState.UNKNOWN_UUID + "-");
+        if (na) {
+            return ClusterState.UNKNOWN_UUID;
```

Comment:
I don't think this is possible for a blob that actually ended up in the repository - we use `_na_` for the index UUID only if the index hasn't been created yet, but in that case we couldn't possibly have snapshotted it.